packages/backend-modules/call-to-actions/graphql/schema-types.js

```diff
@@ -14,11 +14,19 @@ type CallToAction {
   acknowledgedAt: DateTime
 }
 
-type CallToActionPayload {
-  customComponent: CallToActionPayloadCustomComponent!
+union CallToActionPayload = CallToActionComponentPayload | CallToActionBasicPayload
```

Comment:
This change is likely not enough and is a breaking change.

API resolver for CTA payloads needs to provied `__typename`, returning either `CallToActionComponentPayload` or `CallToActionBasicPayload`. Otherwise, API will throw an error as it can't determine what type one returns.

However, not sure how to determine that.

Furthermore www requires to update queries accordingly:

_before_

```gql
query myCallToActions {
  me {
    id
    callToActions {
      id
      beginAt
      endAt
      acknowledgedAt
      updatedAt
      createdAt
      payload {
        customComponent {
          key
          args
        }
      }
    }
  }
}
```

_after_

```gql
query myCallToActions {
  me {
    id
    callToActions {
      id
      beginAt
      endAt
      acknowledgedAt
      updatedAt
      createdAt
      payload {
        ... on CallToActionBasicPayload {
          text
          linkLabel
          linkHref
        }
        ... on CallToActionComponentPayload {
          customComponent {
            key
            args
          }
        }
      }
    }
  }
}
```

Transition is not smooth. It will lead to errors for some users using old www code accessing new API.

This needs a bit more architecting.

Reply:
@jstcki Resolver nicely done.

I assume my comment was misleading. Adding a resolver using `__resolveType` is what resolves `__typename`; not sure why one would want to return an object with `__typename`.

On the breaking change in `query myCallToActions`: That thing is accessed ~30k times a day and I struck me as odd to willfully accept bad requests (until a user is using new www code). But I can accept that as a strategy.

Reply:
@patrickvenetz Ah, okay. I just interpreted it like that because it's actually possible to just return objects with `__typename` and then it works. Apparently many roads lead to a resolved gql type 😂 

Yes, I get that … but if we deploy www + api at the same time it should be fine, I think? But after sleeping on it I think we should actually add a new query and deprecate the old one. WDYT?